docs/DeveloperGuide.md

```diff
@@ -65,10 +65,62 @@ When you are ready to start coding, we recommend that you look at the class diag
 code and the interaction among different classes.<br>
 
 ## Design & implementation
+### Reflection Component
+![Reflection Component Class Diagram](diagrams/ReflectionClassDiagram.png)
+This `Reflection` component provides users with random sets of introspective questions for users to reflect on.<br>
+<br>
+The `reflection` package consists of several classes, namely `ReflectionManager`, `SelfReflection`, `ReflectionQuestion`,
+`GetCommand`, `HomeCommand` and `ReflectUi`. There are three abstract classes `Manager`, `Command`, `TextUi` which some 
+classes in `reflection` package inherit from. But since these abstract classes are outside of `reflection` package, the 
+focus of this section will be on classes inside `reflection` package. <br>
+<br>
+
+`ReflectionManager` class:<br>
+- This class is in charge of the overall execution of the **Self Reflection** feature. 
+- It inherits from abstract class `Manager`
+- Each `ReflectionManager` object contains exactly one `ReflectUi` object as an attribute to get user inputs.
+- The `runEventDriver()` method is the entry of the Self Reflection feature, the caller creates a `ReflectionManager` 
+object and calls this method to launch the Self Reflection feature. It calls a class-level method `SelfReflection.geet()`
+to print greeting logo and message, therefore, `SelfRelfection` class is a dependency of `ReflectionManager` class. 
+- It contains a **while loop** to continuously get input commands from users until users input `home` command to return back to main WellNUS++ interface. 
```

Comment:
It'll be meaningful to explain _why_ a while loop also, e.g. user is expected to continuously give commands until `home` is given